Sming/Makefile-rboot.mk

```diff
@@ -173,31 +179,7 @@ endif
 
 EXTRA_INCDIR += $(SMING_HOME)/include $(SMING_HOME)/ $(LWIP_INCDIR) $(SMING_HOME)/system/include $(SMING_HOME)/Wiring $(SMING_HOME)/Libraries $(SMING_HOME)/SmingCore $(SMING_HOME)/Services/SpifFS $(SDK_BASE)/../include $(THIRD_PARTY_DIR)/rboot $(THIRD_PARTY_DIR)/rboot/appcode $(THIRD_PARTY_DIR)/spiffs/src
 
-USER_LIBDIR  = $(SMING_HOME)/compiler/lib/
```

Comment:
I simply moved it to the same place where it was in `Makefile-project.mk`, but on second analysis it seems that perhaps its location in `Makefile-project.mk` is wrong too, since I see references to `USER_LIBDIR` before it is set. Furthermore, now there's a duplicate line, `# libraries used in this project, mainly provided by the SDK`, which wasn't there before. I'll fix this and move the `USER_LIBDIR` line back to before all its usages, but we should check if `Makefile-project.mk` isn't wrong too.